### PR TITLE
assertMissing and assertHas better handle empty-ish values

### LINKED ISSUE #115: Json::assertHas doesn't work as expected
Hi,

I just updated the library in my project to latest version (1.2) and replaced some places to use `KernelBrowser::browser()->json()->assertHas('selector')`. But it seems that it doesn't work as expected.

I'm testing a json api endpoint which returns this json:
```json
{
  "limit": 50,
  "total": 1,
  "page": 1,
  "pages", 1,
  "_links": {
    "first": {
      "href": "/api/backend/chains?_page=1&_per_page=50"
    },
    "last": {
      "href": "/api/backend/chains?_page=1&_per_page=50"
    }
  }
  "_embedded": {
    "items": [
         {"id": 1, "label": "Grand compte par défaut"}
    ]
  }
}
```

Here is my test which fails:
```php
$this->browser()->get('/uri')->assertStatus(Response::RESPONSE_OK)
    ->json()->assertHas('total');
```

Here is my test which succeeds:
```php
$response = $this->browser()->get('/uri')->assertStatus(Response::RESPONSE_OK)
    ->json()->decoded();
self::assertSame(1, $response['total']);
```

I tried to replace the `Assert::try` in `Json::assertHas` by `Assert::that($selector)->isNotEmpty()` and it works.
I think the problem comes from `length` used which in this case returns something empty as `1` is not countable...
I don't know if my solution is the best to use. If so, let me know I can provide the PR. :-) 

Thank you

### PR DESCRIPTION
**Pull request by flohw**
Closes #115

### CODE REVIEW

File: src/Browser/Json.php [hunk 1]
```diff
@@ -83,15 +86,16 @@ public function assertHas(string $selector): self
...
-        } catch (\RuntimeException $e) {
-            Assert::pass();
+        $prefix = 1 === preg_match('#^\[#', $selector) ? '' : '@.';
+        $exploded = explode('.', $prefix.$selector);
+        $key = array_pop($exploded);
+        $keySelector = implode('.', $exploded);
 
-            return $this;
-        }
+        dump($selector, $keySelector, $key);
```
Comment (kbond) [inline src/Browser/Json.php]: don't forget to remove.

**Review by kbond — CHANGES_REQUESTED** [review]
I'm fine with this. It's not ideal but our hands are a bit tied.

Regarding the tests that fail/are commented out. Not sure how common a requirement this will be. I'm fine leaving as is, then fixing adjusting the logic as demanded by bug reports.

File: src/Browser/Json.php (hunk 1 above)
Comment (flohw) [reply src/Browser/Json.php]: Oups, that's why I waited for feedback on the issue :slightly_smiling_face:

**Comment by flohw** [comment]
Nice, I removed the tests and the forgotten dumps.

Thank you for your feedback. As I said on the other PR, from a certain point of view it's could be logical to assert length on an array and assert(Has|Missing) on objects.
We will se later. This edge case may be a bit hard without real world case.

**Comment by kbond** [comment]
Thanks @flohw!

**Pull request merged**